### PR TITLE
Add methods for `p` output

### PR DESCRIPTION
**Pull request by okkez**

### CODE REVIEW

**Comment by okkez** [comment]
See also https://github.com/fluent/fluentd/wiki/Developer-notes-for-fixes-on-v0.14#configuration

File: lib/fluent/config/section.rb [hunk 1]
```diff
@@ -62,6 +66,19 @@ def [](key)
         @params[key.to_sym]
       end
 
+      def respond_to?(symbol, include_all=false)
+        case symbol
+        when :inspect, :nil?, :to_h, :+, :instance_of?, :kind_of?, :[], :respond_to?, :respond_to_missing?, :method_missing,
+          true
+        when :!, :!= , :==, :equal?, :instance_eval, :instane_exec
```
Comment (uu59) [inline lib/fluent/config/section.rb]: `instane_exec` should be `instance_exec` ?

Comment (okkez) [inline lib/fluent/config/section.rb]: Thanks! I've fixed.

**Comment by tagomoris** [comment]
Can you add some tests for this change?

**Comment by okkez** [comment]
OK! I will add tests.

**Comment by okkez** [comment]
I've added tests.

**Comment by tagomoris** [comment]
@okkez tests are failing with syntax error on Ruby 2.0 & 2.1.

File: test/config/test_section.rb [hunk 2]
```diff
@@ -106,6 +106,42 @@ class TestSection < ::Test::Unit::TestCase
...
+        sub_test_case '#to_s' do
+          test '#to_s == #inspect' do
+            h1 = {name: "s1", num: 10, class: "A"}
+            s1 = Fluent::Config::Section.new(h1)
+
+            assert_equal(s1.to_s, s1.inspect)
+          end
+        end
+
+        data("inspect": [:inspect, true],
```
Comment (cosmo0920) [inline test/config/test_section.rb]: {"string": value} JSON syntax has been introduced in Ruby 2.3.
But Fluentd v0.14 targets Ruby 2.1 or later.
We should use {"string" => value} JSON syntax.

Comment (okkez) [inline test/config/test_section.rb]: Good catch. I will fix this.

**Comment by okkez** [comment]
I've fixed tests.

**Comment by tagomoris** [comment]
LGTM.

**Comment by tagomoris** [comment]
Merged. Thank you!

**Pull request merged**